apps/challenges/aws_utils.py

```diff
@@ -697,3 +699,109 @@ def restart_workers_signal_callback(sender, instance, field_name, **kwargs):
                 instance.pk, field_name
             )
         )
+
+
+def create_eks_nodegroup(instance, cluster_name):
+    """
+    Creates a nodegroup when a EKS cluster is created by the EvalAI admin
+    """
+    nodegroup_name = "{}-nodegroup".format(instance.title.replace(" ", "-"))
+    client = get_boto3_client("eks", aws_keys)
+    # TODO: Move the hardcoded cluster configuration such as the instance_type, subnets, AMI to challenge configuration later.
+    try:
+        response = client.create_nodegroup(
+            clusterName=cluster_name,
+            nodegroupName=nodegroup_name,
+            scalingConfig={"minSize": 1, "maxSize": 10, "desiredSize": 1},
+            diskSize=100,
+            subnets=[VPC_DICT["SUBNET_1"], VPC_DICT["SUBNET_2"], ],
+            instanceTypes=["g4dn.xlarge"],
+            amiType="AL2_x86_64_GPU",
+            nodeRole=settings.EKS_NODEGROUP_ROLE_ARN,
+        )
+    except ClientError as e:
+        logger.exception(e)
+        return response
+    waiter = client.get_waiter("nodegroup_active")
+    waiter.wait(
+        clusterName=cluster_name,
+        nodegroupName=nodegroup_name,
+    )
+    # TODO:Deploy nvidia deamonset
+
+
+def create_eks_cluster(sender, instance, field_name, **kwargs):
```

Comment:
How about replacing `instance` with `challenge` or something that explains it, as it only is used as `Challenge`'s instance.